shared/stack/stack_service.go

```diff
@@ -0,0 +1,430 @@
+package stack
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	backend "github.com/chanzuckerberg/happy/shared/backend/aws"
+	"github.com/chanzuckerberg/happy/shared/diagnostics"
+	"github.com/chanzuckerberg/happy/shared/model"
+	"github.com/chanzuckerberg/happy/shared/options"
+	"github.com/chanzuckerberg/happy/shared/util"
+	workspacerepo "github.com/chanzuckerberg/happy/shared/workspace_repo"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-tfe"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
+	"golang.org/x/sync/errgroup"
+)
+
+type StackServiceIface interface {
+	NewStackMeta(stackName string) *StackMeta
+	Add(ctx context.Context, stackName string, options ...workspacerepo.TFERunOption) (*Stack, error)
+	Remove(ctx context.Context, stackName string, options ...workspacerepo.TFERunOption) error
+	GetStacks(ctx context.Context) (map[string]*Stack, error)
+	GetStackWorkspace(ctx context.Context, stackName string) (workspacerepo.Workspace, error)
+}
+
+type StackService struct {
+	backend       *backend.Backend
+	workspaceRepo workspacerepo.WorkspaceRepoIface
+	executor      util.Executor
+	env, appName  string
+
+	// NOTE: creator Workspace is a workspace that creates dependent workspaces with
+	// given default values and configuration
+	// the derived workspace is then used to launch the actual happy infrastructure
+	creatorWorkspaceName string
+}
+
+func NewStackService(env, appName string) *StackService {
+	return &StackService{
+		executor: util.NewDefaultExecutor(),
+		env:      env,
+		appName:  appName,
+	}
+}
+
+func (s *StackService) GetWritePath() string {
+	return fmt.Sprintf("/happy/%s/stacklist", s.env)
```

Comment:
This needs to be pushed down to the implementation - k8s_compute_backend or ecs_compute_backend.

Reply:
agreed, but I copied this code from the CLI so that it could be leverage by the API. Should I fix this now or should I make another PR to fix it?